pyscript.core/src/stdlib/pyweb/pydom.py

```diff
@@ -0,0 +1,420 @@
+            if self._element.type == "checkbox":
+                self._element.checked = value
+            elif self._element.type == "number":
+                self._element.value = float(value)
+            else:
+                self._element.value = value
+        else:
+            self._element.innerHTML = value
+
+    def clear(self):
+        self.value = ""
+
+    def clone(self, new_id=None):
+        clone = Element(self._element.cloneNode(True))
+        clone.id = new_id
+
+        return clone
+
+    def remove_class(self, classname):
+        classList = self._element.classList
+        if isinstance(classname, list):
+            classList.remove(*classname)
+        else:
+            classList.remove(classname)
+        return self
+
+    def add_class(self, classname):
+        classList = self._element.classList
+        if isinstance(classname, list):
+            classList.add(*classname)
+        else:
+            self._element.classList.add(classname)
+        return self
+
+    @property
+    def classes(self):
+        classes = self._element.classList.values()
+        return [x for x in classes]
+
+    def show_me(self):
+        self._element.scrollIntoView()
+
+    def when(self, event, handler):
+        document.when(event, selector=self)(handler)
+
+    # @staticmethod
+    # def when(element, event_type):
+    #     # TODO: Ideally, we should have that implemented in PyScript not patched here
+    #     # if isinstance(element, Element):
+    #     #     element = [element]
+    #     def decorator(func):
+    #         # elements = js.document.querySelectorAll(selector)
+    #         sig = inspect.signature(func)
+
+    #         # Function doesn't receive events
+    #         if not sig.parameters:
+
+    #             def wrapper(*args, **kwargs):
+    #                 func()
+
+    #             # for el in element:
+    #             add_event_listener(element._element, event_type, wrapper)
+    #         else:
+    #             # for el in element:
+    #             add_event_listener(element._element, event_type, func)
+    #         return func
+    #     return decorator
+
+
+class StyleProxy(dict):
+    def __init__(self, element: Element) -> None:
+        self._element = element
+
+    @cached_property
+    def _style(self):
+        return self._element._element.style
+
+    def __getitem__(self, key):
+        self._style[key]
+
+    def __setitem__(self, key, value):
+        self._style.setProperty(key, value)
+
+    def pop(self, key):
+        self._style.removeProperty(key)
+
+    def set(self, **kws):
+        for k, v in kws.items():
+            self._element._element.style.setProperty(k, v)
+
+    # CSS Properties
+    # Reference: https://github.com/microsoft/TypeScript/blob/main/src/lib/dom.generated.d.ts#L3799C1-L5005C2
+    # Following prperties automatically generated from the above reference using
+    # tools/codegen_css_proxy.py
+    @property
```

Comment:
this looks like it's screaming for a Proxy ... we're adding only a few random properties instead of enabling them all with less lines of code? can't we have a proper proxy instead via getters/setters? result would be the same for these properties, but every other property would be allowed out of the box.

Reply:
`_style` is the proxy. Proxying all the style properties to the top-level object would cancel the point of "a small and pythonic API" and also not be very well supported by editors (although we could create an interface file).

Reply:
> would cancel the point of "a small and pythonic API"

I am not sure I am following ... I am suggesting to use less code to offer a complete Pythonic solution

Reply:
Let's hop on a call soon and make sure we are on the same page :)

Reply:
I have implemented `style` proxy myself ... it's a proxy attribute in W3C standards too, it's about having a proxy that translates operations with it into DOM observable changes ...

```js
pyelement.style.background_color = "#ABC"
```

that should translate into `realelement.style.backgroundColor = "#ABC"` and return that value when asked ... this makes pydom automatically compatible with all CSS properties one can define in the Web world.